src/helpers/qownnotesmarkdownhighlighter.cpp

```diff
@@ -55,12 +55,9 @@ QOwnNotesMarkdownHighlighter::~QOwnNotesMarkdownHighlighter()
     delete wordTokenizer;
 }
 
-void QOwnNotesMarkdownHighlighter::updateCurrentNote(Note *_note) {
-    if (_note == nullptr) {
-        _currentNote = Note::fetch(qApp->property("currentNoteId").toInt());
```

Comment:
Why did you remove these lines? They are pretty important. :grin:

Reply:
This function is only being called once for every note from the `setNoteTextFromNote` function. The note can never be null, and using a ptr here seemed useless.

The null check is now being performed in the `setNoteTextFromNote` for safety.

As for the second line, it's pretty much useless if we pass the note directly to `_currentnote`, no?

Reply:
Phew, I just saw you disabled the call from `QOwnNotesMarkdownHighlighter::highlightBlock` in `Call updateCurrentNote() only once when setting the note (#1362)`. That maybe will break note linking and maybe other things in certain circumstances...

Reply:
I guess we'll wait and see.

Reply:
> Phew, I just saw you disabled the call from `QOwnNotesMarkdownHighlighter::highlightBlock` in `Call updateCurrentNote() only once when setting the note (#1362)`. That maybe will break note linking and maybe other things in certain circumstances...

Yes that was done a while ago.

It doesn't need to be called for every block since the note is always same. But when the note changes we update the currentNote().
Calling it in highlightBlock is very expensive.